### PR TITLE
MFB-788: Fix 3 QA failures in WA Apple Health Medicaid calculator

### PR DESCRIPTION
**Pull request by elliottwaissbluth**
## Summary

- Override `member_value()` in `WaAppleHealthMedicaid` to fix 3 eligibility mismatches found during [MFB-788](https://linear.app/myfriendben/issue/MFB-788) QA testing (8/11 passing → 11/11). All three failures were caused by PolicyEngine not modeling WA-specific or federal rules that require custom calculator logic.
- Add **foster care categorical Medicaid** — children ≤20 with `fosterChild` relationship are categorically eligible regardless of household income ([42 U.S.C. § 1396a(a)(10)(A)(i)(I)](https://www.law.cornell.edu/uscode/text/42/1396a); WAC 182-505-0211)
- Add **Medicare exclusion for ACA expansion** — Medicare-entitled adults cannot use the expansion pathway ([42 CFR § 435.119(b)(3)](https://www.ecfr.gov/current/title-42/section-435.119)); seniors/disabled route to ABD, others return ineligible
- Add **premium CHIP tier fallback** — uninsured children <19 above the free tier (215% FPL) but below the premium ceiling (317% effective FPL) qualify for Apple Health for Kids with a premium (WAC 182-505-0215), using hardcoded HCA 2026 income thresholds

## QA failures addressed

| Scenario | Issue | Root cause |
|----------|-------|------------|
| 4 — Medicare adult | Incorrectly shown as eligible ($5,652) | PE ignores `insurance.medicare` flag for expansion pathway |
| 8 — Child at premium tier | Incorrectly shown as ineligible ($0) | PE only models free tier; no concept of WA's premium CHIP extension |
| 11 — Foster care child | Incorrectly shown as ineligible ($0) | PE has no foster care categorical pathway; applies standard income test |

## Test plan

- [x] 27 new unit tests in `TestWaAppleHealthMedicaid` covering all three pathways, boundary conditions, and wiring — all passing
- [x] Ruff lint and format clean
- [ ] After deploy to staging: re-run `/api-qa-execution MFB-788 staging` to confirm all 11 scenarios pass end-to-end

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Release Notes

* **New Features**
  * Enhanced Washington Apple Health Medicaid eligibility calculations with improved routing for foster children, seniors, and disabled individuals
  * Added premium CHIP coverage tier support for uninsured children based on household income thresholds

* **Tests**
  * Expanded test coverage for new eligibility pathways and income ceiling calculations

[Review Change Stack](https://app.coderabbit.ai/change-stack/MyFriendBen/benefits-api/pull/1546?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: programs/programs/wa/pe/tests/test_member.py [hunk 1]
```diff
@@ -15,12 +19,344 @@
...
+        """HH1 ceiling matches the hardcoded HCA 2026 value."""
+        self.assertEqual(WaAppleHealthMedicaid._premium_chip_ceiling(1), 50_591)
+
+    def test_premium_chip_ceiling_hh8_extrapolated(self):
+        """HH8 is extrapolated: HH7 base + 1 × additional-member increment."""
+        expected = 158_627 + 18_006
+        self.assertEqual(WaAppleHealthMedicaid._premium_chip_ceiling(8), expected)
+
+    def test_premium_chip_ceiling_hh10_extrapolated(self):
+        """HH10 is extrapolated: HH7 base + 3 × additional-member increment."""
```
[the remark this answers is not in the document]
Comment (elliottwaissbluth) [reply programs/programs/wa/pe/tests/test_member.py:350]: fixed in update

File: programs/programs/wa/pe/member.py [hunk 2]
```diff
@@ -23,6 +24,85 @@ class WaAppleHealthMedicaid(Medicaid):
...
+                    dependency.member.MedicaidSeniorOrDisabled, member.id
+                )
+                if not qualifies:
+                    return 0
+                if is_disabled:
+                    return self.medicaid_categories["DISABLED"] * 12
+                return self.medicaid_categories["AGED"] * 12
+
+            # Medicare-entitled, not senior/disabled → ineligible for expansion
+            return 0
```
[the remark this answers is not in the document]
Comment (elliottwaissbluth) [reply programs/programs/wa/pe/member.py]: fixed in update
